SampleApps/ReceiverDemoSwift/DPLReceiverSwiftAppDelegate.swift

```diff
@@ -6,29 +6,45 @@ class DPLReceiverSwiftAppDelegate: UIResponder, UIApplicationDelegate {
     var window: UIWindow?
     lazy var router: DPLDeepLinkRouter = DPLDeepLinkRouter()
 
-    func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
         
-        // Register a class to a route using object subscripting
-        self.router["/product/:sku"] = DPLProductRouteHandler.self
+        // Register a block to a route (matches dpl:///product/93598)
+        self.router.registerRoute("/product/:sku") { link in
+            
+            if let rootViewController = application.keyWindow?.rootViewController as? UINavigationController {
+                if let storyboard = rootViewController.storyboard {
+                    if let controller = storyboard.instantiateViewController(withIdentifier: "detail") as? DPLTargetViewController {
+                        controller.configure(with: link)
+                        rootViewController.pushViewController(controller as! UIViewController, animated: false)
+                    }
+                }
+            }
+        }
         
-        // Register a class to a route using the explicit registration call
+        self.router.registerRoute("/log/:message") { link in
+            if let link = link {
```

Comment:
Can we make the link passed to the block non-optional?

Reply:
Yes. That's the plan but at the moment we don't have any nullability annotations and adding them technically breaks the API. I'd rather do this in 2.0. 

The focus of this change is to disable subscript registration in Swift because the blocks were ignored internally and never called. It looks like you can't even use that API in Swift 4.